src/runtime/policy/policy.ts

```diff
@@ -0,0 +1,270 @@
+/**
+ * @license
+ * Copyright (c) 2020 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+import * as AstNode from '../manifest-ast-nodes.js';
+import {AnnotationRef} from '../recipe/annotation.js';
+import {flatMap} from '../util.js';
+
+export enum PolicyEgressType {
+  Logging = 'Logging',
+  FederatedAggregation = 'FederatedAggregation',
+}
+
+export enum PolicyRetentionMedium {
+  Ram = 'Ram',
+  Disk = 'Disk',
+}
+
+export enum PolicyAllowedUsageType {
+  Any = 'Any',
+  Egress = 'Egress',
+  Join = 'Join',
+}
+
+/**
+ * Definition of a dataflow policy.
+ *
+ * This class is just a thin wrapper around the Policy AST node. There's no
+ * actual functionality other than validating the input is correct.
+ */
+export class Policy {
+  constructor(
+      readonly name: string,
+      readonly targets: PolicyTarget[],
+      readonly configs: PolicyConfig[],
+      private readonly annotationRefs: AnnotationRef[]) {
+    this.validateEgressType();
+  }
+
+  get description(): string | null {
+    const description = this.annotationRefs.find(a => a.name === 'intendedPurpose');
```

Comment:
nit - variable should be `intendedPurpose` for consistency with the annotation name, and avoiding chicken.chicken.
another nit - my personal preference is keeping getters really simple and doing all this work in ctor. up to you though.

Reply:
Done